lib/hets-agent/subscriber.rb

```diff
@@ -57,12 +59,20 @@ def subscribe_worker_queue(requirement)
       return unless version_requirement_satisfied?(requirement)
       queue = create_worker_queue(requirement)
       print_listening(requirement) if print?
-      queue.subscribe(block: false, manual_ack: true,
+      queue.subscribe(block: false,
+                      manual_ack: true,
                       timeout: 0) do |delivery_info, _properties, body|
-        response = call_hets(JSON.parse(body))
-        if response&.status&.zero?
-          queue.channel.acknowledge(delivery_info.delivery_tag)
-        end
+        $stderr.puts "received job: #{body}" if print?
+        handle_job(queue, delivery_info, body)
+      end
+    end
+
+    def handle_job(queue, delivery_info, body)
+      response = call_hets(JSON.parse(body))
+      if response&.status&.zero?
+        queue.channel.acknowledge(delivery_info.delivery_tag)
+      else
+        queue.channel.reject(delivery_info.delivery_tag)
```

Comment:
What status code does hets return if it e.g. failed parsing a file? Because in that case we probably don't want to reject the job, since that would mean re-queueing it to be parsed again.